Convenience constructors for constant (nearest-neighbor) interpolation

eirikbrandsaas opened the pull request:
Unless there is some specific reason for this not be in the code, I don't see why this shouldn't be included.

If you want to merge the pull request, I'd be happy to update the docs and tests.

mkitti commented:
Seems reasonable. Add some tests please.

Also, perhaps we would like to support `Previous`, `Nearest`, and `Next` in these convenience constructors.
https://github.com/JuliaMath/Interpolations.jl/blob/master/src/b-splines/constant.jl#L4-L6

eirikbrandsaas commented:
I added tests for regular and irregular grids for 1d and 2d constant interpolation.

I also updated the docs (in two separate commits).

The first 08b0ee5 just adds constant interpolations to the docs and lists it as working with irregular grids.
The second f454c7d adds a motivating example on the use of the convenience constructor (because it wasn't that obvious to me at first).

I didn't add the previous or next, because a) I don't need them (sorry!) and b) maybe they should be added in a more clever way (e.g. with an optional argument for using the different constant methods).

mkitti commented:
Wonderful, thank you.
